src/command_graph_generator.cc

```diff
@@ -425,7 +418,13 @@ void command_graph_generator::generate_await_pushes(batch& current_batch, const
 }
 
 void command_graph_generator::update_local_buffer_fresh_regions(const task& tsk, const std::unordered_map<buffer_id, region<3>>& per_buffer_local_writes) {
-	auto requirements = get_buffer_requirements_for_mapped_access(tsk, subrange<3>(tsk.get_global_offset(), tsk.get_global_size()), tsk.get_global_size());
+	buffer_requirements_list requirements;
+	for(const auto bid : tsk.get_buffer_access_map().get_accessed_buffers()) {
+		buffer_requirements reqs{bid, {}, {}};
+		reqs.consumed = tsk.get_buffer_access_map().get_task_consumed_region(bid);
+		reqs.produced = tsk.get_buffer_access_map().get_task_produced_region(bid);
+		requirements.emplace_back(std::move(reqs));
```

Comment:
```suggestion
		const auto& tm = tsk.get_buffer_access_map();
		requirements.emplace_back(bid, tm.get_task_consumed_region(bid), tm.get_task_produced_region(bid));
```

Reply:
Done!

Reply:
...I had to switch to `push_back` because Clang 14 doesn't support `emplace_back` for structs w/o constructor.